core/kernel/persistence/smoothsql/class.SmoothModel.php

```diff
@@ -73,6 +73,18 @@ function getProperty($uri)
         return $property;
     }
 
+    public function isWritable(core_kernel_classes_Resource $resource): bool
+    {
+        /** @var core_kernel_classes_Triple $triple */
+        foreach ($resource->getRdfTriples() as $triple) {
+            if (!in_array((int)$triple->modelid, $this->getWritableModels(), true)) {
```

Comment:
Please, store `$this->getWritableModels()` in a separate variable and place it outside the `foreach`-statement.